osu.Game/Screens/Play/Player.cs

```diff
@@ -600,8 +595,9 @@ private IBeatmap loadPlayableBeatmap(Mod[] gameplayMods, CancellationToken cance
         /// Whether the pause or fail dialog should be shown before performing an exit.
         /// If <see langword="true"/> and a dialog is not yet displayed, the exit will be blocked and the relevant dialog will display instead.
         /// </param>
+        /// <param name="withoutTransition">Whether the exit should perform without a transition, because the screen had faded to black already.</param>
         /// <returns>Whether this call resulted in a final exit.</returns>
-        protected bool PerformExit(bool showDialogFirst)
+        protected bool PerformExit(bool showDialogFirst, bool withoutTransition = false)
```

Comment:
Does this second bool param need to exist? There's it, there's `noExitTransition` the private field, and the `quickRestart` params everywhere else. The naming is confusing, and the data flow seems to be overly convoluted - why aren't the callers of this setting `withoutTransition` directly to whatever they wish it to be?

Reply:
I wanted it to only be set when the `Exit()` call occurs, which can often be cancelled halfway through the `PerformExit` flow...